src/pages/ReportChangeApproverPage.tsx

```diff
@@ -0,0 +1,106 @@
+import React, {useCallback, useMemo, useState} from 'react';
+import {View} from 'react-native';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import RenderHTML from '@components/RenderHTML';
+import ScreenWrapper from '@components/ScreenWrapper';
+import SelectionList from '@components/SelectionList';
+import RadioListItem from '@components/SelectionList/RadioListItem';
+import Text from '@components/Text';
+import useEnvironment from '@hooks/useEnvironment';
+import useLocalize from '@hooks/useLocalize';
+import useOnyx from '@hooks/useOnyx';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {assignCurrentUserAsApprover} from '@libs/actions/Report';
+import Navigation from '@libs/Navigation/Navigation';
+import type {PlatformStackScreenProps} from '@libs/Navigation/PlatformStackNavigation/types';
+import type {ReportChangeApproverParamList} from '@libs/Navigation/types';
+import {getManagerAccountEmail, isMemberPolicyAdmin, isPolicyAdmin} from '@libs/PolicyUtils';
+import {isMoneyRequestReport, isMoneyRequestReportPendingDeletion} from '@libs/ReportUtils';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import NotFoundPage from './ErrorPage/NotFoundPage';
+import type {WithReportOrNotFoundProps} from './home/report/withReportOrNotFound';
+import withReportOrNotFound from './home/report/withReportOrNotFound';
+
+type ReportChangeApproverPageProps = WithReportOrNotFoundProps & PlatformStackScreenProps<ReportChangeApproverParamList, typeof SCREENS.REPORT_CHANGE_APPROVER.ROOT>;
+
+function ReportChangeApproverPage({report, policy, isLoadingReportData}: ReportChangeApproverPageProps) {
+    const reportID = report?.reportID;
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const {environmentURL} = useEnvironment();
+
+    const [session] = useOnyx(ONYXKEYS.SESSION, {canBeMissing: false});
+    const [selectedApproverType, setSelectedApproverType] = useState<string>();
+
+    const changeApprover = useCallback(() => {
+        if (!selectedApproverType) {
+            return;
+        }
+
+        if (!isPolicyAdmin(policy) || !policy || !session?.accountID) {
+            return;
+        }
+        assignCurrentUserAsApprover(report);
```

Comment:
If we already have `session?.accountID` maybe we pass that to `assignCurrentUserAsApprover` directly? I think this is the best practice to make these functions less dependent on global Onyx data (though not widely followed).